libr/anal/sign.c

```diff
@@ -774,6 +767,109 @@ R_API bool r_sign_add_anal(RAnal *a, const char *name, ut64 size, const ut8 *byt
 	return retval;
 }
 
+static RSignGraph *r_sign_fcn_graph(RAnalFunction *fcn) {
+	r_return_val_if_fail (fcn, false);
+	RSignGraph *graph = R_NEW0 (RSignGraph);
+	if (graph) {
+		graph->cc = r_anal_function_complexity (fcn),
+		graph->nbbs = r_list_length (fcn->bbs);
+		graph->edges = r_anal_function_count_edges (fcn, &graph->ebbs);
+		graph->bbsum = r_anal_function_realsize (fcn);
+	}
+	return graph;
+}
+
+static RSignBytes *r_sign_fcn_bytes(RAnal *a, RAnalFunction *fcn) {
+	r_return_val_if_fail (a && fcn, false);
+	RCore *core = a->coreb.core;
+	int maxsz = a->coreb.cfggeti (core, "zign.maxsz");
+	int fcnlen = r_anal_function_realsize (fcn);
+	int size = R_MIN (core->io->addrbytes * fcnlen, maxsz);
+
+	// hopefully not necissary, but just in case
+	if (size <= 0) {
+		eprintf ("error: Negative size for bytes signature\n");
+		return NULL;
+	}
+
+	// get bytes part of signature
+	ut8 *bytes = malloc (size);
+	if (!bytes) {
+		return NULL;
+	}
+	if (r_io_nread_at (core->io, fcn->addr, bytes, size) < 0) {
```

Comment:
check must be == size or take the result and use it to avoid analyzing invalid data